packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -77,36 +78,63 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
             result(FlutterMethodNotImplemented)
         }
     }
-
+    
     private func onConfigureDataStore(args: [String: Any], result: @escaping FlutterResult) {
-
+        
         guard let modelSchemaList = args["modelSchemas"] as? [[String: Any]] else {
             result(false)
             return //TODO
         }
-
+        
+        let syncExpressionList = args["syncExpressions"] as! [[String: Any]]
         let syncInterval = args["syncInterval"] as? Double ?? DataStoreConfiguration.defaultSyncInterval
         let syncMaxRecords = args["syncMaxRecords"] as? UInt ?? DataStoreConfiguration.defaultSyncMaxRecords
         let syncPageSize = args["syncPageSize"] as? UInt ?? DataStoreConfiguration.defaultSyncPageSize
-
+        let controller = UIApplication.shared.windows.first { $0.isKeyWindow }!.rootViewController as! FlutterViewController
+        let channel = FlutterMethodChannel(name: "com.amazonaws.amplify/datastore", binaryMessenger: controller.binaryMessenger)
+        
         do {
-
+            
+            
             let modelSchemas: [ModelSchema] = try modelSchemaList.map {
                 try FlutterModelSchema.init(serializedData: $0).convertToNativeModelSchema()
             }
-
+            
             modelSchemas.forEach { (modelSchema) in
                 flutterModelRegistration.addModelSchema(modelName: modelSchema.name, modelSchema: modelSchema)
             }
-
+            
+            let syncExpressions: [DataStoreSyncExpression] = try syncExpressionList.map { syncExpression in
```

Comment:
This `onConfigureDataStore` became quite long now, would that be better to extract the synchronous query predicate resolving logic to a dedicated function? Same for the Android implementation.

Reply:
Yes, excellent suggestion!